### ISSUE TITLE
CMake compilation options for VC++

### ISSUE THREAD

**Issue by MikesAracade** [Cylix/cpp_redis#13]
I tried using CMake to build the Visual Studio 2015 project files for cpp_redis.

I ran into an issue when trying to compile using the generated project.

This line:
**## set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -W -Wall -Wextra -O3")**

is not valid for Visual Studio 2015.
I don't know a thing about gcc compiler so I'll I tried to find documentation to convert these flags for VC++.

It complains about the -W flag.
I believe these options are what you want for VC++
**/W3 - Warning Level 3 (any lower produces errors)
/O2 - Creates fast code (Don't want this for Debug build)**
There is no option for c++11 (it's on by default)
Which would give you the following CMake line for a Release build.

**## set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -W3 -O2 -D "_UNICODE" -D "UNICODE" -D "WIN32_LEAN_AND_MEAN")**

I have no clue how to use CMake or to put in conditional statements for the environment.
Can you merge this fix in?
For reference here's a web link to VC++ compiler flags: https://msdn.microsoft.com/en-us/library/fwkeyyhe.aspx

Also could you make the default project a Static Library as opposed to a DLL (or maybe support both)?
Lastly the 'tests" project is not automatically built by CMake.

FYI I'll create a branch for the Windows port of cpp_redis.  This library is exactly what I've been looking for!
Mike M.

**Comment by Cylix** [comment]
Hi,

I added some changes on the branch `windows_cmake`.
I have set the following flags for the Windows platform:

```cmake
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W3 /O2 -D '_UNICODE' -D 'UNICODE' -D 'WIN32_LEAN_AND_MEAN'")
```

I also now compile the library as a static library, not a shared one.
Indeed, there was no reason to compile this as a shared library!

I do not have a configured Windows environment right now, so I haven't been able to try these changes. 
I'm going to setup a Windows environment as soon as I can and will merge this onto master once I have a confirmation it is fine.

However, if you have the time to try this on your environment and can confirm me it works as expected, I might be able to merge this faster or to make the appropriate modification if necessary (see branch `windows_cmake`).

Concerning the "tests" project not built automatically by CMake: this is intentional.
I guess that most people that want to use the library are not willing to build the tests.
As described in the [wiki](https://github.com/Cylix/cpp_redis/wiki/Install#building-tests), tests are only built if it has been explicitly specified to CMake.

Thanks again for your interest in cpp_redis and for your time, I really do appreciate a lot!
